pxd_fastpath.c

```diff
@@ -624,23 +635,15 @@ static struct pxd_io_tracker* __pxd_init_block_replica(struct pxd_device *pxd_de
 	atomic_set(&iot->active, 0);
 	atomic_set(&iot->fails, 0);
 	iot->file = fileh;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,8,0)
-	iot->read = (bio_op(bio) == REQ_OP_READ);
-#else
 	iot->read = (bio_data_dir(bio) == READ);
-#endif
-
-	pxd_printk("pxd %p:__pxd_init_block_replica clone bio %p (blkg %p), resetting backing block dev to %p\n",
-			pxd_dev, clone_bio, clone_bio->bi_blkg, bdi);
 
+	clone_bio->bi_private = pxd_dev;
 	if (S_ISBLK(inode->i_mode)) {
 		BIO_SET_DEV(clone_bio, bdi);
+		clone_bio->bi_end_io = pxd_complete_io;
+	} else {
+		clone_bio->bi_end_io = pxd_complete_io_dummy;
```

Comment:
I'm not sure this is helpful. Why not just fail the op?

Reply:
I dont follow your comment. If the IO is for a backed up file (i.e. file io, then bi_end_io should never be called). This field gets used only for block io. This just initializes the bio_end_io to a dummy callback, which catches any inconsistent callbacks if ever received, which should be never.

Reply:
If this can never happen why setup the dummy callback at all? If it's an error condition return err to the caller.

Reply:
to catch any issues in our fuse driver. This is not a end user error condition, this will be a px fuse driver error condition.